pkgs/_analyzer_macros/lib/analyzer_macro_implementation.dart

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:_macro_host/macro_host.dart';
+// ignore: implementation_imports
+import 'package:analyzer/src/summary2/macro_injected_impl.dart' as injected;
+import 'package:dart_model/dart_model.dart';
+import 'package:macro_service/macro_service.dart';
+import 'package:macros/macros.dart';
+// ignore: implementation_imports
+import 'package:macros/src/executor.dart' as injected;
+
+/// Injected macro implementation for the analyzer.
+class MacroImplementation implements injected.MacroImplementation, HostService {
+  final Uri packageConfig;
+  final Map<String, String> macroImplByName;
```

Comment:
I can infer the keys are names (probably class names), but I don't know what the values are. Can you add a comment describing this a bit?

Reply:
It's described in the comment where it's passed in to `start`: value are fully qualified name of the implementation.

This will anyway go away in the next PR when the placeholder pubspec reader is used :)